client/signup/steps/site-or-domain/index.jsx

```diff
@@ -3,47 +3,31 @@ import { Gridicon } from '@automattic/components';
 import { SelectItems } from '@automattic/onboarding';
 import { globe, addCard, layout } from '@wordpress/icons';
 import { localize } from 'i18n-calypso';
-import { get, isEmpty } from 'lodash';
 import { Component } from 'react';
 import { connect } from 'react-redux';
-import QueryProductsList from 'calypso/components/data/query-products-list';
 import { getUserSiteCountForPlatform } from 'calypso/components/site-selector/utils';
 import { domainRegistration } from 'calypso/lib/cart-values/cart-items';
 import { getDomainProductSlug } from 'calypso/lib/domains';
 import { preventWidows } from 'calypso/lib/formatting';
 import StepWrapper from 'calypso/signup/step-wrapper';
 import { isUserLoggedIn, getCurrentUser } from 'calypso/state/current-user/selectors';
-import { getAvailableProductsList } from 'calypso/state/products-list/selectors';
 import { submitSignupStep } from 'calypso/state/signup/progress/actions';
 
 import './style.scss';
 
 class SiteOrDomain extends Component {
 	getDomainName() {
-		const { signupDependencies } = this.props;
-		let isValidDomain = false;
-		const domain = get( signupDependencies, 'domainItem.meta', false );
-
-		if ( domain ) {
-			if ( domain.split( '.' ).length > 1 ) {
-				const productSlug = getDomainProductSlug( domain );
-				isValidDomain = !! this.props.productsList[ productSlug ];
```

Comment:
This was basically checking if the domain has a TLD that we sell in WPCOM, which should already be done in the shopping cart now.